javascript/app-framework/src/components/resources/FileTree/FileTree.types.ts

```diff
@@ -7,16 +7,48 @@ import type { DataShape } from '@datashaper/schema/dist/datatable/DataShape.js'
 import type { TableContainer } from '@datashaper/tables'
 import type { BaseFile } from '@datashaper/utilities'
 
+import type { DataShaperAppPlugin } from '../../../index.js'
+
 export interface FileTreeProps {
+	/**
+	 * The static CSS style to apply to the outer container
+	 */
 	style?: React.CSSProperties
+
+	/**
+	 * The class attribute to specify on the outer container
+	 */
 	className?: string
-	selectedFileId?: string
+
+	/**
+	 * Whether the file-tree pane is expanded (full-width)
+	 */
 	expanded: boolean
+
+	/**
+	 * Toggle the file-tree expansion
+	 */
 	toggleExpanded: () => void
-	examples?: FileDefinition[]
-	appResources: ResourceTreeData[]
-	selectedKey?: string
+
+	/**
+	 * Examples to present to the user in the command-bar menu
+	 */
+	examples: FileDefinition[]
+
+	/**
+	 * A map of profile-name to tree-node generator
+	 */
+	plugins: Map<string, DataShaperAppPlugin>
```

Comment:
Maybe not critical ATM, but it would be really nice to have a tree component where these types of plugin concerns are decomposed to custom render functions or contextual menu items, rather than domain-aware